src/main/java/io/cdap/plugin/github/source/common/SchemaBuilder.java

```
@@ -0,0 +1,89 @@
+    List<Schema.Field> schemaFields = new ArrayList<>();
+    for (Field field : fields) {
+      Schema.Type schemaType = getSchemaType(field.getType().getSimpleName());
+      if (schemaType.isSimpleType()) {
+        schemaFields.add(getSimpleSchemaField(field));
+      } else if (Schema.Type.ARRAY.equals(schemaType)) {
+        ParameterizedType parameterizedType = (ParameterizedType) field.getGenericType();
+        Class<?> clazz = (Class<?>) parameterizedType.getActualTypeArguments()[0];
+        Schema schema = buildSchema(field.getName(), clazz);
+        schemaFields.add(Schema.Field.of(field.getName(), Schema.arrayOf(schema)));
+      } else {
+        Schema schema = buildSchema(field.getName(), field.getType());
+        schemaFields.add(Schema.Field.of(field.getName(), schema));
+      }
+    }
+    return Schema.recordOf(schemaName, schemaFields);
+  }
+
+  public static List<Field> getModelFields(Class<?> clazz) {
+    List<Field> fields = new ArrayList<>();
+    Class currentClass = clazz;
+    while (currentClass != Object.class) {
+      Field[] declaredFields = currentClass.getDeclaredFields();
+      fields.addAll(Arrays.asList(declaredFields));
+      currentClass = currentClass.getSuperclass();
+    }
+    return fields;
+  }
+
+  private static Schema.Field getSimpleSchemaField(Field field) {
+    return Schema.Field.of(field.getName(), Schema.nullableOf(Schema.of(
+      getSchemaType(field.getType().getSimpleName()))));
+  }
+
+  private static Schema.Type getSchemaType(String className) {
+    switch (className) {
+      case "String": {
+        return Schema.Type.STRING;
+      }
+      case "Integer": {
+        return Schema.Type.INT;
+      }
+      case "Long": {
+        return Schema.Type.LONG;
+      }
+      case "Boolean": {
+        return Schema.Type.BOOLEAN;
+      }
+      case "List": {
+        return Schema.Type.ARRAY;
+      }
+      default: {
+        return Schema.Type.RECORD;
```

why is it a default? It should fail in any unsupported case, not fallback to RECORD type.

---

this method supports all types, when passing a type argument, an object is configured from it

---

I wrote unit test for it